src/GraphQL/Validation/DocumentValidator.cs

```diff
@@ -79,22 +93,20 @@ public class DocumentValidator : IDocumentValidator
             }
             else if (!rules.Any())
             {
-                return SuccessfullyValidatedResult.Instance;
+                var ctx = Initialize();
+                var variables = ctx.GetVariableValues(schema, operation?.Variables, inputs);
```

Comment:
No variable visitors here.

Reply:
I think that I liked `GetVariableValues` back in `ExecutionHelper` better.  But it's okay either way.

Reply:
We should consider that with document caching, now a validation context is required even though no rules are executed (correct?).  👎   At least with context reuse this can be minimized.

Reply:
> I think that I liked GetVariableValues back in ExecutionHelper better

I need access to the `ValidationContext` to be able to call `ReportError`. Code from `ExecutionHelper` just throws exceptions.

> now a validation context is required even though no rules are executed (correct?)

Yes. This was a necessary measure because it was necessary to have access to `ReportError`. In principle, it would be possible to leave the methods static, but then it would require working with a large number of variables in order to accumulate errors.

Reply:
As indicated elsewhere, I do not think `GetVariableValue` should call `ReportError` at all.  Exceptions should be handled within `GetVariableValues` and `ReportError` handled from there, so any error will immediately invalidate the entire variable.

I think `GetVariableValue` can return to being a static method, not that it matters much.

As a side effect, this allows us the option of modifying/cloning `GetVariableValues` (which is not much code) so that it can be used without a `ValidationContext`.

Reply:
> Exceptions should be handled within GetVariableValues

Handled _how_ ? `try/catch` with capturing throwed exception into `ReportError` call ? It doesn't look very good.

Reply:
I think it’s easier to understand than having every method having Boolean return values and out parameters. It’s very clear what’s going on. An error occurs. It is caught and added to the validation error list — exactly the same way that would happen for a field resolver.

Reply:
OK. Done. 

> so that it can be used without a `ValidationContext`

Nope, eventually `IVariableVisitor` needs `ValidationContext`.